app/src/main/java/com/keylesspalace/tusky/components/notifications/NotificationFetcher.kt

```diff
@@ -35,10 +36,12 @@ class NotificationFetcher @Inject constructor(
                     val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
 
                     // Create sorted list of new notifications
-                    val notifications = fetchNewNotifications(account)
-                        .filter { filterNotification(notificationManager, account, it) }
-                        .sortedWith(compareBy({ it.id.length }, { it.id })) // oldest notifications first
-                        .toMutableList()
+                    val notifications = runBlocking { // OK, because in a worker thread
```

Comment:
Still a huge code smell imho. What speaks against making `fetchAndShow` suspending and `NotificationWorker` a `CoroutineWorker` instead?

Reply:
Nothing. That's on my list of things to tackle after https://github.com/tuskyapp/Tusky/pull/3649 lands (which is what I did for the new PruneCacheWorker in that PR).

I'm trying to avoid work that's unrelated to getting v22 released. That's currently blocking a large queue of open PRs